src/ctapipe/irf/benchmarks.py

```diff
@@ -0,0 +1,306 @@
+) -> tuple[QTable, np.ndarray, tuple[int, int]]:
+    result = QTable()
+    result["ENERG_LO"], result["ENERG_HI"] = split_bin_lo_hi(
+        e_bins[np.newaxis, :].to(u.TeV)
+    )
+    result["THETA_LO"], result["THETA_HI"] = split_bin_lo_hi(
+        fov_bins[np.newaxis, :].to(u.deg)
+    )
+    fov_bin_index, _ = calculate_bin_indices(events["true_source_fov_offset"], fov_bins)
+    mat_shape = (len(fov_bins) - 1, len(e_bins) - 1)
+    return result, fov_bin_index, mat_shape
+
+
+class EnergyBiasResolutionMakerBase(DefaultTrueEnergyBins):
+    """
+    Base class for calculating the bias and resolution of the energy prediciton.
+    """
+
+    def __init__(self, config=None, parent=None, **kwargs):
+        super().__init__(config=config, parent=parent, **kwargs)
+
+    @abstractmethod
+    def __call__(
+        self, events: QTable, extname: str = "ENERGY BIAS RESOLUTION"
+    ) -> BinTableHDU:
+        """
+        Calculate the bias and resolution of the energy prediction.
+
+        Parameters
+        ----------
+        events: astropy.table.QTable
+            Reconstructed events to be used.
+        extname: str
+            Name of the BinTableHDU.
+
+        Returns
+        -------
+        BinTableHDU
+        """
+
+
+class EnergyBiasResolution2dMaker(EnergyBiasResolutionMakerBase, DefaultFoVOffsetBins):
+    """
+    Calculates the bias and the resolution of the energy prediction in bins of
+    true energy and fov offset.
+    """
+
+    def __init__(self, config=None, parent=None, **kwargs):
+        super().__init__(config=config, parent=parent, **kwargs)
+
+    def __call__(
+        self, events: QTable, extname: str = "ENERGY BIAS RESOLUTION"
+    ) -> BinTableHDU:
+        result, fov_bin_idx, mat_shape = _get_2d_result_table(
+            events=events,
+            e_bins=self.true_energy_bins,
+            fov_bins=self.fov_offset_bins,
+        )
+        result["N_EVENTS"] = np.zeros(mat_shape)[np.newaxis, ...]
+        result["BIAS"] = np.full(mat_shape, np.nan)[np.newaxis, ...]
+        result["RESOLUTION"] = np.full(mat_shape, np.nan)[np.newaxis, ...]
+
+        for i in range(len(self.fov_offset_bins) - 1):
+            bias_resolution = energy_bias_resolution(
+                events=events[fov_bin_idx == i],
+                energy_bins=self.true_energy_bins,
+                bias_function=np.mean,
+                energy_type="true",
+            )
+            result["N_EVENTS"][:, i, :] = bias_resolution["n_events"]
+            result["BIAS"][:, i, :] = bias_resolution["bias"]
+            result["RESOLUTION"][:, i, :] = bias_resolution["resolution"]
+
+        return BinTableHDU(result, name=extname)
+
+
+class AngularResolutionMakerBase(DefaultTrueEnergyBins, DefaultRecoEnergyBins):
+    """
+    Base class for calculating the angular resolution.
+    """
+
+    # Use reconstructed energy by default for the sake of current pipeline comparisons
+    use_true_energy = Bool(
```

Comment:
I'd invert the option. `use_reco_energy` with default false. The default should be the "good" option.